libs/editor/egui_editor/src/android/keyboard.rs

```diff
@@ -0,0 +1,676 @@
+            231 => Some(VoiceAssist),
+            232 => Some(TvRadioService),
+            233 => Some(TvTeletext),
+            234 => Some(TvNumberEntry),
+            235 => Some(TvTerrestrialAnalog),
+            236 => Some(TvTerrestrialDigital),
+            237 => Some(TvSatellite),
+            238 => Some(TvSatelliteBs),
+            239 => Some(TvSatelliteCs),
+            240 => Some(TvSatelliteService),
+            241 => Some(TvNetwork),
+            242 => Some(TvAntennaCable),
+            243 => Some(TvInputHdmi1),
+            244 => Some(TvInputHdmi2),
+            245 => Some(TvInputHdmi3),
+            246 => Some(TvInputHdmi4),
+            247 => Some(TvInputComposite1),
+            248 => Some(TvInputComposite2),
+            249 => Some(TvInputComponent1),
+            250 => Some(TvInputComponent2),
+            251 => Some(TvInputVga1),
+            252 => Some(TvAudioDescription),
+            253 => Some(TvAudioDescriptionMixUp),
+            254 => Some(TvAudioDescriptionMixDown),
+            255 => Some(TvZoomMode),
+            256 => Some(TvContentsMenu),
+            257 => Some(TvMediaContextMenu),
+            258 => Some(TvTimerProgramming),
+            259 => Some(Help),
+            260 => Some(NavigatePrevious),
+            261 => Some(NavigateNext),
+            262 => Some(NavigateIn),
+            263 => Some(NavigateOut),
+            264 => Some(StemPrimary),
+            265 => Some(Stem1),
+            266 => Some(Stem2),
+            267 => Some(Stem3),
+            268 => Some(DpadUpLeft),
+            269 => Some(DpadDownLeft),
+            270 => Some(DpadUpRight),
+            271 => Some(DpadDownRight),
+            272 => Some(MediaSkipForward),
+            273 => Some(MediaSkipBackward),
+            274 => Some(MediaStepForward),
+            275 => Some(MediaStepBackward),
+            276 => Some(SoftSleep),
+            277 => Some(Cut),
+            278 => Some(Copy),
+            279 => Some(Paste),
+            280 => Some(SystemNavigationUp),
+            281 => Some(SystemNavigationDown),
+            282 => Some(SystemNavigationLeft),
+            283 => Some(SystemNavigationRight),
+            284 => Some(AllApps),
+            285 => Some(ThumbsUp),
+            286 => Some(ThumbsDown),
+            287 => Some(ProfileSwitch),
+            _ => None,
+        }
+    }
+
+    pub fn valid_text(&self) -> bool {
+        use AndroidKeys::*;
+        match self {
+            A | B | C | D | E | F | G | H | I | J | K | L | M | N | O | P | Q | R | S | T | U
+            | V | W | X | Y | Z | Key0 | Key1 | Key2 | Key3 | Key4 | Key5 | Key6 | Key7 | Key8
+            | Key9 | Apostrophe | Backslash | Comma | Equals | LeftBracket | Minus | Period
+            | RightBracket | Semicolon | Space => true,
+
+            _  => false,
+        }
+    }
+
+    pub fn egui_key(&self) -> Option<egui::Key> {
+        use AndroidKeys::*;
+        let key = match self {
```

Comment:
I belive you're missing Backspace and End

Reply:
You're right thanks for the catch.